packages/mcfunction/test/parser/macro.spec.ts

```diff
@@ -0,0 +1,24 @@
+import {
+	showWhitespaceGlyph,
+	testParser,
+} from '@spyglassmc/core/test-out/utils.js'
+import { describe, it } from 'mocha'
+import snapshot from 'snap-shot-it'
+import { macro } from '../../lib/parser/index.js'
+
+describe('mcfunction parser macro()', () => {
+	const cases: { content: string }[] = [
+		{ content: '$say $(message)' },
+		{ content: '$tag @s add $(tag)_40' },
+		{ content: '$scoreboard players set $mode settings $(value)' },
+		{ content: '$tellraw $(player) $(text)' },
+		{ content: '$tellraw $(players) ["$ Hello everyone! $"]' },
+		{ content: '$summon cow $(x).1 $(y)30.0 $(z).0' },
+	]
```

Comment:
this is great! nice job 😄 

we should also add at least one case for each type of error we can report so far.

something like this should cover all our bases:

```ts
{ content: '$say $(foo bar' },
{ content: '$say $() bar' },
{ content: '$say $(invalid.key) bar' },
{ content: '$say no macro argument specified' },
```

Reply:
the motivation behind this is so that we can be confident we don't have a regression in functionality if, for whatever reason, we need to modify the `macro` parser in the future (not that I expect us to need to)